.github/workflows/build-release.yml

```diff
@@ -0,0 +1,68 @@
+name: Build and Release
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    # Check all PR
+
+jobs:
+  build:
+    name: Build distribution
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.12"
+    - run: python -m pip install tox
+    - name: Build package
+      run: tox -e build
+      env:
+        # Use the CPU only version of torch when building/running the code
+        PIP_EXTRA_INDEX_URL: https://download.pytorch.org/whl/cpu
```

Comment:
This should not be required here, since the Python side handles using a CUDA-enabled build on a CPU machine.

Reply:
Ahh really? This means we don't need this at all in the CI?

Reply:
yep, this is only required if you are building C++ code with CMake that links to torch

Reply:
Good to know!